workspaces/optic-engine/src/interactions/result.rs

```diff
@@ -97,17 +97,50 @@ impl UnmatchedRequestUrl {
 
 #[derive(Clone, Debug, Deserialize, Serialize, Hash)]
 #[serde(rename_all = "camelCase")]
-pub struct UnmatchedQueryParameters {
+pub struct UnmatchedQueryParametersDescriptor {
   pub interaction_trail: InteractionTrail,
   pub requests_trail: RequestSpecTrail,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, Hash)]
+#[serde(untagged)]
+pub enum UnmatchedQueryParameters {
+  Observed(UnmatchedQueryParametersDescriptor),
+
+  #[serde(skip)]
+  Unobserved(UnmatchedQueryParametersDescriptor),
+}
+
```

Comment:
It was hard to find an exact way to express this new diff result variant. In a way there being no query parameters documented and no query parameters in an interaction, you can say that _matches_. However, all `Match*` variants so far all describe cases where we match something seen with something that exists in the spec, not both sides not existing. In the end, what we're willing to capture in both diffs and the learner is that we don't have a documented query parameters yet.

Going with "observed" and "unobserved" variants of the same main variant was the best I could come up with so far, keeping the close relationship between the two results in place.